File: bbarena-model/src/main/java/net/sf/bbarena/model/event/game/EndDriveEvent.java
package net.sf.bbarena.model.event.game;

import net.sf.bbarena.model.Arena;
import net.sf.bbarena.model.util.Concat;

public class EndDriveEvent extends GameEvent {

	private static final long serialVersionUID = -1919966604189789283L;
    private Integer _oldDrive;

    public EndDriveEvent() {
        super();
	}

	@Override
	public void doEvent(Arena arena) {
		_arena = arena;

        _oldDrive = _arena.getMatch().getCurrentDrive();
        _arena.getMatch().setCurrentDrive(_oldDrive + 1);
    }

	@Override
	protected void undoEvent() {
        _arena.getMatch().setCurrentDrive(_oldDrive);
    }

	@Override
	public String getString() {
		return Concat.buildLog(getClass());
	}

}

File: src/main/kotlin/bbarena/model/event/game/EndDriveEvent.kt
package bbarena.model.event.game

import bbarena.model.Arena
import bbarena.model.util.Concat

class EndDriveEvent(var oldDrive: Int = 0) : GameEvent() {

    override val string: String
        get() = Concat.buildLog(javaClass)

    override fun doEvent(arena: Arena) {
        oldDrive = arena.match?.currentDrive!!
        arena.match.currentDrive = oldDrive + 1
    }

    override fun undoEvent(arena: Arena) {
        arena.match?.currentDrive = oldDrive
    }

}
